Mage.Sets/src/mage/cards/k/KjeldoranEliteGuard.java

```diff
@@ -0,0 +1,128 @@
+ */
+public class KjeldoranEliteGuard extends CardImpl {
+
+    public KjeldoranEliteGuard(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{3}{W}");
+        this.subtype.add(SubType.HUMAN);
+        this.subtype.add(SubType.SOLDIER);
+        this.power = new MageInt(2);
+        this.toughness = new MageInt(2);
+
+        // Target creature gets +2/+2 until end of turn.
+        // When that creature leaves the battlefield this turn, sacrifice Kjeldoran Elite Guard.
+        // Activate only during combat.
+        Ability ability = new ConditionalActivatedAbility(
+                Zone.BATTLEFIELD,
+                new KjeldoranEliteGuardEffect(),
+                new TapSourceCost(),
+                new IsPhaseCondition(TurnPhase.COMBAT, false));
+        ability.addTarget(new TargetControlledCreaturePermanent());
+        this.addAbility(ability);
+    }
+
+    private KjeldoranEliteGuard(final KjeldoranEliteGuard card) { super(card); }
+
+    @Override
+    public Card copy() {
+        return new KjeldoranEliteGuard(this);
+    }
+}
+
+class KjeldoranEliteGuardEffect extends OneShotEffect {
+
+    KjeldoranEliteGuardEffect() {
+        super(Outcome.Neutral);
+        staticText = "Target creature gets +2/+2 until end of turn. "
+                + "When that creature leaves the battlefield this turn, sacrifice Kjeldoran Elite Guard. "
+                + "Activate only during combat.";
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        if (source == null) { return false; }
+
+        Permanent targetCreature = game.getPermanent(source.getFirstTarget());
+        if (targetCreature == null) { return false; }
+
+        // Target creature gets +2/+2 until end of turn.
+        BoostTargetEffect buffEffect = new BoostTargetEffect(2, 2, Duration.EndOfTurn);
+        buffEffect.setTargetPointer(new FixedTarget(source.getFirstTarget(), game));
+        game.addEffect(buffEffect, source);
+
+        // When that creature leaves the battlefield this turn, sacrifice Kjeldoran Elite Guard.
+        SacrificeTargetEffect sacrificeKjeldoranEliteGuardEffect = new SacrificeTargetEffect();
+        sacrificeKjeldoranEliteGuardEffect.setTargetPointer(new FixedTarget(source.getSourceId(), game));
+        game.addDelayedTriggeredAbility(
+                new KjeldoranEliteGuardDelayedTriggeredAbility(
+                        sacrificeKjeldoranEliteGuardEffect,
+                        source.getFirstTarget()),
+                source);
+
+        return true;
+    }
+
+    private KjeldoranEliteGuardEffect(KjeldoranEliteGuardEffect effect) { super(effect); }
+
+    @Override
+    public KjeldoranEliteGuardEffect copy() { return new KjeldoranEliteGuardEffect(this); }
+}
+
+class KjeldoranEliteGuardDelayedTriggeredAbility extends DelayedTriggeredAbility {
+
+    UUID creatureId;
+
+    KjeldoranEliteGuardDelayedTriggeredAbility(Effect effect, UUID creatureId) {
```

Comment:
you don't need to pass the effect through the constructor here since it's always gonna be the same

Reply:
What do I pass to super then since they all require an effect?

Reply:
you just pass `new SacrificeSourceEffect()`

Reply:
I see. So delete 79-80 above, and create the sacrificeEffect directly inside the constructor.

Reply:
yeah, exactly